neqo-transport/tests/network.rs

```diff
@@ -148,6 +149,147 @@ simulate!(
+            for loss in &[0, 1, 2, 5, 10, 20] {
+                Simulator::new(
+                    format!("diff conditions upload {} {} {}", b, *delay, *loss),
+                    boxed![
+                        ConnectionNode::new_client(boxed![
+                            ReachState::new(State::Confirmed),
+                            SendData::new(TRANSFER_AMOUNT_LARGE),
+                            ReachState::new(State::Closed(ConnectionError::Transport(
+                                Error::IdleTimeout
+                            )))
+                        ]),
+                        Drop::percentage(*loss),
+                        TailDrop::new(10 * 1 << 20, 32_768, Duration::from_millis(*delay)),
+                        ConnectionNode::new_server(boxed![
+                            ReachState::new(State::Confirmed),
+                            ReceiveData::new(TRANSFER_AMOUNT_LARGE),
+                            ReachState::new(State::Closed(ConnectionError::Transport(
+                                Error::IdleTimeout
+                            )))
+                        ]),
+                        TailDrop::new(10 * 1 << 20, 32_768, Duration::from_millis(*delay)),
+                    ],
+                )
+                .run();
+            }
+        }
+    }
+}
+
+// bandwidth 10Mbps, 20Mbps, 50Mbps and 100Mbps
+// delay by 10ms, 50ms, 100ms and 250ms
+// loss 0%
+// With tail-drop queue
+#[test]
+fn diff_conditions_client_download_tail_drop() {
+    for b in &[10, 20, 50, 100] {
+        for delay in &[10, 25, 50, 100, 250] {
+            Simulator::new(
+                format!("diff conditions download taildrop only {} {}", b, *delay),
+                boxed![
+                    ConnectionNode::new_client(boxed![
+                        ReachState::new(State::Confirmed),
+                        ReceiveData::new(TRANSFER_AMOUNT_LARGE),
+                        ReachState::new(State::Closed(ConnectionError::Transport(
+                            Error::IdleTimeout
+                        )))
+                    ]),
+                    TailDrop::new(10 * 1 << 20, 32_768, Duration::from_millis(*delay)),
+                    ConnectionNode::new_server(boxed![
+                        ReachState::new(State::Confirmed),
+                        SendData::new(TRANSFER_AMOUNT_LARGE),
+                        ReachState::new(State::Closed(ConnectionError::Transport(
+                            Error::IdleTimeout
+                        )))
+                    ]),
+                    TailDrop::new(10 * 1 << 20, 32_768, Duration::from_millis(*delay)),
+                ],
+            )
+            .run();
+        }
+    }
+}
+
+// bandwidth 10Mbps, 20Mbps, 50Mbps and 100Mbps
+// delay by 10ms, 50ms, 100ms and 250ms
+// loss 0%
+// With tail-drop queue
+#[test]
+fn diff_conditions_client_upload_taildrop() {
+    for b in &[10, 20, 50, 100] {
+        for delay in &[10, 25, 50, 100, 250] {
```

Comment:
You don't seem to be using `b` here other than in debugging.  Also, with 20 different configurations, the large transfer is going to take a long time to run.